docs/network-drivers.md

```diff
@@ -1,47 +1,127 @@
+    vlans = 300-500, 700-750
+
+# Network allocator drivers
+
+The only network allocator shipped with HaaS that is of interest to
+users (there are others useful for development purposes) is the VLAN
+allocator. The name of the extension is
+``haas.ext.network_allocators.vlan_pool``, and it requires a single
+extension-specific config option, `vlans`, which is a comma separated
+list of VLAN ids and/or ranges of VLAN ids. Networks created within HaaS
+will use VLANs specified in the configuration file. An example::
+
+    ...
+
+    [extensions]
+    haas.ext.network_allocators.vlan_pool =
+    ...
+
+    [haas.ext.network_allocators.vlan_pool]
+    vlans = 300, 500-700, 800-950
+    ...
+
+# Switch drivers
+
+At present, all switch drivers shipped with HaaS require that the VLAN
+pool allocator is in use. There are two switch drivers shipped with
+HaaS:
+
+* ``haas.ext.switches.dell``, which provides a driver for the Dell
+  Powerconnect 5500 series switches.
+* ``haas.ext.switches.nexus``, which provides a driver for some Cisco
+  Nexus switches. Only the 3500 and 5500 have been tested, though it is
+  possible that other models will work as well.
+
+Neither driver requires any extension-specific config options. Per the
+information in `rest_api.md`, the details of certain API calls are
+driver-dependant, below are the details for each of these switches.
+
+## Powerconnect driver
+
+### switch_register
+
+To register a Dell Powerconnect switch, the ``"type"`` field of the
+request body must have a value of::
+
+    http://schema.massopencloud.org/haas/switches/powerconnect55xx
+
+In addition, it requires three extra fields: ``"username"``,
+``"hostname"``, and ``"password"``, which provide the necessary
+information to connect to the switch via telnet (``"hostname"`` may also
+be an IP address).  SSH support is planned, but even so we do not
+recommend allowing connectivity to a switch's management interface from
+an untrusted network.
+
+### switch_register_port
+
+Port names must be of the same form accepted by the switch's console
+interface, e.g. ``gi1/0/5``. Be *very* careful when specifying these, as
+they are not validated by HaaS (this will be fixed in future versions).
+
+## Nexus driver
+
+### switch_register
+
+The type field for the Nexus driver has the value::
+
+    http://schema.massopencloud.org/haas/switches/nexus
+
+The nexus driver requires the same additional fields as the powerconnect
+driver, plus an additional field "dummy_vlan", which should be a JSON
+number corresponding to an unused VLAN id on the switch. This VLAN
+should be deactivated (and thus no traffic should flow across it ever).
+This exists to get around an implementation problem related to disabling
+the native VLAN.
+
+### switch_register_port
+
+Like the powerconnect driver, the Nexus driver accepts port names of the
+same format accepted by the underlying switch, in this case (e.g.)
+``ethernet 1/42``. The same concerns about validation apply.
```

Comment:
Just a suggestion which you can take or leave: it might be worth rewording this section without the powerconnect references, since the reader may skip straight to this section. As we add drivers (even 1 or 2 more), it could become cumbersome to both manage and navigate references to other drivers.